datafusion/physical-plan/src/union.rs

```diff
@@ -260,6 +260,13 @@ impl ExecutionPlan for UnionExec {
     fn benefits_from_input_partitioning(&self) -> Vec<bool> {
         vec![false; self.children().len()]
     }
+
+    fn supports_limit_pushdown(&self) -> bool {
+        // UnionExec is supporting limit_pushdown because it's only applicable with
+        // SortPreservingMergeExec plan. Since SortExec does not support limit_pushdowm
+        // it's not possible to lose sorting in UnionExec's children.
```

Comment:
We discussed adding the supports_limit_pushdown method to UnionExec or not. We agreed to do so because LimitPushDown is only applicable to SPM in UnionExec's possible children. That's why I put this comment actually, maybe we can put a better comment I don't know